src/code.cloudfoundry.org/cf-tcp-router/config/config.go

```diff
@@ -120,63 +125,43 @@ func (c *Config) initConfigFromFile(path string, enableCertCreation bool) error
 		c.DrainWaitDuration = DrainWaitDefault
 	}
 
-	if enableCertCreation && len(c.FrontendTLSJob) > 0 {
-		owner, err := user.Lookup("root")
-		if err != nil {
-			return err
-		}
-
-		uid, err := strconv.Atoi(owner.Uid)
-		if err != nil {
-			return err
-		}
-
-		gid, err := resolveGroupID("vcap", "root")
-		if err != nil {
-			return err
-		}
+	if len(c.FrontendTLSJob) > 0 {
```

Comment:
Shouldn't there be an overarching if for `f enableCertCreation` ? And there should be an error case if `len(c.FrontendTLSJob) == 0`, which I assume is an invalid state that would not be currently caught.

Reply:
Based on [this code](https://github.com/cloudfoundry/routing-release/blob/401e03290261d0c5bd387cb04a3cba71f7b1a401/jobs/tcp_router/templates/pre-start.erb#L4). Certs will only be added to the file system on the canary VM.

Is that intended?

Reply:
Regarding the different contexts - if we attempt to write the certs during the normal bootup, the following error causes the job to fail:

```
==> /var/vcap/sys/log/tcp_router/tcp_router.stdout.log <==
Wed Sep  3 04:09:33 AM UTC 2025: Starting HAProxy
Wed Sep  3 04:09:33 AM UTC 2025: Successfully started HAProxy
uaa.service.cf.internal has address 88.0.0.31
{"timestamp":"2025-09-03T04:09:33.466539294Z","level":"info","source":"tcp-router","message":"tcp-router.starting","data":{}}
{"timestamp":"2025-09-03T04:09:33.476675165Z","level":"error","source":"tcp-router","message":"tcp-router.failed-to-unmarshal-config-file","data":{"error":"frontend_tls[0]: Error changing ownership: chown /var/vcap/jobs/tcp_router/config/certs/tcp-router/frontend: read-only file system"}}
```

Reply:
Spoke offline.

> To write during runtime you can do it in the data dir and setup a writeable path via BPM [here](https://github.com/cloudfoundry/routing-release/blob/9956cbcb36fe0c80f404871fbd4782b6655ce414/jobs/tcp_router/templates/bpm.yml.erb#L12-L19). For example "/var/vcap/data/tcp_router/config" is already writable.

Decision was made to create a directory inside `/var/vcap/data/tcp_router/config` which holds the certificates and to be written on bootup